CONTRIBUTING.md

```diff
@@ -243,18 +243,82 @@ If in doubt, you can always ask for guidance in the Pull Request or on
 [IRC in the #node-dev channel](https://webchat.freenode.net?channels=node-dev&uio=d4).
 
 Feel free to post a comment in the Pull Request to ping reviewers if you are
-awaiting an answer on something.
+awaiting an answer on something. If you encounter words or acronyms that
+seem unfamiliar, check out the [glossary](https://sites.google.com/a/chromium.org/dev/glossary).
 
+Note that your commits usually get squashed when they are landed (see
+notes about [commit squashing](#commit-squashing)).
 
 ### Step 8: Landing
 
-Once your Pull Request has been reviewed and approved by at least one Node.js
-Collaborators (often by saying LGTM, or Looks Good To Me), and as long as
-there is consensus (no objections from a Collaborator), a
-Collaborator can merge the Pull Request . GitHub often shows the Pull Request as
- `Closed` at this point, but don't worry. If you look at the branch you raised
- your Pull Request against (probably `master`), you should see a commit with
- your name on it. Congratulations and thanks for your contribution!
+In order to get landed, a Pull Request needs to be reviewed and
+[approved](#getting-approvals-for-your-pull-request) by
+at least one Node.js Collaborator. If this Pull Request touches
+more than the documentation, then it also needs an all-green
+[CI (Continuous Integration) test run](#ci-testing). After that,
+as long as there is no objections from a Collaborator, this
+Pull Request can be merged. If you find your Pull Request waiting
+longer than you expect, see
+[notes about the waiting time](#waiting-until-the-pull-request-gets-landed).
+
+When the Pull Request is landed, the person who land this will post
+a comment to the Pull Request page mentioning the commit(s) in which
+it is landed. GitHub often shows the Pull Request as `Closed` at this
+point, but don't worry. If you look at the branch you raised your
+Pull Request against (probably `master`), you should see a commit with
+your name on it. Congratulations and thanks for your contribution!
+
+## Additional Notes
+
+### Commit Squashing
+
+When the commits in your Pull Request get landed, they will be squashed
+into one commit per logical change, with metadata added to the commit
+message (including links to the Pull Request, links to relevant issues,
+and the names of the reviewers). As long as you don't delete
+your fork branch, the commit history of your Pull Request will stay
+intact on the Pull Request page.
```

Comment:
It's just a reminder telling people if they delete their fork branch, the commit history will be lost, should other people want to see it later. Sometimes the commit history might be useful if the PR is big.